### ISSUE TITLE
Arrowhead anti-pattern in examples

### ISSUE THREAD

**Issue by SantiagoTorres** [secure-systems-lab/code-style-guidelines#3]
I think we should avoid [arrowhead anti-patterns](https://blog.codinghorror.com/flattening-arrow-code/) as much as possible. I do see that they are put all throughout the suggestions. While this doesn't explicitly say "do arrowhead code," it invites new programmers to do it.

We should update the specification to avoid this.

Update: After re-reading, it seems that this happens mostly [here](https://github.com/secure-systems-lab/code-style-guidelines#never-use-short-circuit-evaluation-to-produce-output) and it's not as widespread as I thought.

**Comment by JustinCappos** [comment]
Sure, please propose an edit and we can add this.

**Comment by lukpueh** [comment]
@SantiagoTorres, I just overhauled the style guide in #21 and removed the recommendation that has the problematic example you mentioned.

I did open a discussion about wether we really want to remove the recommendation (https://github.com/secure-systems-lab/code-style-guidelines/pull/21#discussion_r526068466). But others want to keep it, we definitely have to revise the example to not recommend the arrowhead anti-pattern.

FYI, I also added a note about that anti-pattern in another example that [recommends the "return early"](https://github.com/secure-systems-lab/code-style-guidelines/blame/49b4aa89d72ad014d909bbef055ab534f58ec2db/python.md#L199-L200) pattern.
